api/server/router/network/network_routes.go

```diff
@@ -599,8 +585,8 @@ func (n *networkRouter) findUniqueNetwork(term string) (types.NetworkResource, e
 		}
 	}
 	if len(listByPartialID) > 1 {
-		return types.NetworkResource{}, fmt.Errorf("network %s is ambiguous (%d matches found based on ID prefix)", term, len(listByPartialID))
+		return types.NetworkResource{}, errdefs.InvalidParameter(errors.Errorf("network %s is ambiguous (%d matches found based on ID prefix)", term, len(listByPartialID)))
 	}
 
-	return types.NetworkResource{}, libnetwork.ErrNoSuchNetwork(term)
+	return types.NetworkResource{}, errdefs.NotFound(libnetwork.ErrNoSuchNetwork(term))
```

Comment:
Same warning does not apply here I assume? (re: `libnetwork.ErrNoSuchNetwork()` being special)

Reply:
It doesn't seem like it since nothing should be consuming this other than the HTTP API.
In any case the caller can use `errors.Cause()`